etcdserver/raft.go

```diff
@@ -223,6 +226,8 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 				// gofail: var raftAfterSave struct{}
 
 				if !raft.IsEmptySnap(rd.Snapshot) {
+					// waits etcd server to finish renaming snap db to db.
+					<-replaceDBDone
```

Comment:
it seems to me that the purpose of `raftDone` is to notified etcdserver that raft has persisted log.

The purpose of `replaceDBDone` is to notify raft that etcdserver has rename the snapshot db to db.

Both chans serve different purposes. I don't see a good way to control above logic with one chan.